src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/ConnectionString.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Rest.ClientRuntime.Azure.TestFramework
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// This class represents the connection string being set by the user
+    /// e.g. TEST_CSM_ORGID_AUTHENTICATION="AADTenant=72f98AAD-86f1-2d7cd011db47;ServicePrincipal=72f98AAD-86f1-2d7cd011db47;Password=tzT2+LJBRkSAursui7/Qgo+hyQQ=;SubscriptionId=5562fbd2-HHHH-WWWW-a55d-lkjsldkjf;BaseUri=https://management.azure.com/;AADAuthEndpoint=https://login.windows.net/;GraphUri=https://graph.windows.net/"
+    /// </summary>
+    public class ConnectionString
+    {
+        #region fields
+        private Dictionary<string, string> _keyValuePairs;
+        private string _connString;
+        private StringBuilder _parseErrorSb;
+        #endregion
+
+        #region Properties
+
+        private bool CheckViolation { get; set; }
+
+        /// <summary>
+        /// Represents key values pairs for the parsed connection string
+        /// </summary>
+        public Dictionary<string,string> KeyValuePairs
+        {
+            get
+            {
+                if(_keyValuePairs == null)
+                {
+                    _keyValuePairs = new Dictionary<string, string>();
+                }
+                return _keyValuePairs;
+            }
+            private set
+            {
+                _keyValuePairs = value;
+            }
+        }
+        
+        /// <summary>
+        /// Returns all the parse errors while parsing connection string
+        /// </summary>
+        public string ParseErrors
+        {
+            get
+            {
+                if(_parseErrorSb == null)
```

Comment:
If you intialize the StringBuilder at construction time, tese checks should not be necessary